src/main/java/pl/simpleascoding/tutoringplatform/api/AdvertisementController.java

```diff
@@ -20,7 +20,8 @@ public class AdvertisementController {
     private final AdvertisementService advertisementService;
 
     @PostMapping("/add")
-    public ResponseEntity<RscpDTO<AdvertisementDTO>> createAdvertisement(@RequestBody CreateAdvertisementDTO requestDTO) {
-        return new ResponseEntity<>(advertisementService.createAdvertisement(requestDTO), HttpStatus.CREATED);
+    public ResponseEntity<AdvertisementDTO> createAdvertisement(@RequestBody CreateAdvertisementDTO requestDTO) {
+        RscpDTO<AdvertisementDTO> serviceResult = advertisementService.createAdvertisement(requestDTO);
+        return new ResponseEntity<>(serviceResult.body(), HttpStatus.resolve(serviceResult.status().value()));
```

Comment:
Please split this line into:
```java
        RscpDTO<AdvertisementDTO> rscpDTO= advertisementService.createAdvertisement(requestDTO);

                 // Body
                 AdvertisementDTO advDTO = rscpDTO.body();

                 // Message
                 HttpHeaders headers = new HttpHeaders();
                 headers.add("message", rscpDTO.message());

                 // Status
                 RscpStatus rscpStatus = serviceResult.status().value();
                 HttpStatus httpStatus = HttpStatus.resolve(rscpStatus);

                 return new ResponseEntity<>(advDTO, headers, httpStatus);
```

[Related info](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html#ResponseEntity-T-org.springframework.util.MultiValueMap-org.springframework.http.HttpStatus-)